sklearn/datasets/_twenty_newsgroups.py

```diff
@@ -391,23 +394,39 @@ def fetch_20newsgroups_vectorized(*, subset="train", remove=(), data_home=None,
 
         .. versionadded:: 0.22
 
+    as_frame : bool, default=False
+        If True, the data is a pandas DataFrame including columns with
+        appropriate dtypes (numeric, string or categorical). The target is
+        a pandas DataFrame or Series depending on the number of target_columns.
+
+        .. versionadded:: 0.24
+
     Returns
     -------
     bunch : :class:`~sklearn.utils.Bunch`
         Dictionary-like object, with the following attributes.
 
         data: sparse matrix, shape [n_samples, n_features]
             The data matrix to learn.
+            If ``as_frame`` is True, ``data`` is a pandas object.
         target: array, shape [n_samples]
             The target labels.
         target_names: list, length [n_classes]
             The names of target classes.
+            If ``as_frame`` is True, ``target`` is a pandas object.
```

Comment:
probably deserves a `.. versionchanged` directive

Reply:
here I feel the `versionchanged` is misleading, as `target_names` isn't added in v0.24. The directive has already been added to the new `as_frame` kwarg, so it should be unambiguous to users that this line belongs to that addition.